lib/odoo.js

```diff
@@ -38,11 +38,17 @@ class Odoo {
 		};
 
 		return got.post(`${config.host}:${config.port}/web/session/authenticate`, {json: true, headers, body})
-			.then(res => new OdooClient(res.body.result, {
-				host: config.host,
-				port: config.port,
-				sid: res.headers['set-cookie'][0].split(';')[0]
-			}));
+			.then(res => {
+				if (res.body.error) {
+					// HTTP Status is 200 / OK!
+					throw new Error(res.body.error.message);
```

Comment:
It looks like Ava's `throws()` doesn't like a `Promise` that is rejected with a `string`:
https://github.com/avajs/ava/blob/master/test/promise.js#L213

Personally I like to use `Error` because it captures the stack trace at the moment it is created.

What's you motivation for wanting `string` here?